src/Native/Unix/System.Native/pal_process.c

```diff
@@ -339,7 +339,21 @@ int32_t SystemNative_ForkAndExecProcess(const char* filename,
     // on another thread while a vfork() child is still pending, bad things are possible; however we
     // do not do that.
 
+#if defined (__GLIBC__)
     if ((processId = vfork()) == 0) // processId == 0 if this is child process
+#else
+    // musl libc has an undocumented failure mode around setuid(); we must exclude it.
+    if (setCredentials)
+    {
+        processId = fork();
+    }
+    else
+    {
+        processId = vfork();
```

Comment:
this could be `processId = setCredentials ? fork() : vfork()`

Reply:
That's exactly what the comment I was asked to remove said to not do. I don't trust it.